File: examples/spark-scala/src/main/scala/org/andre/mlflow/examples/MLflowUtils.scala
package org.andre.mlflow.examples

import scala.collection.JavaConversions._
import org.mlflow.tracking.MlflowClient

object MLflowUtils {
  def getOrCreateExperimentId(client: MlflowClient, experimentName: String) : Long = {
    val expOpt = client.listExperiments() find (_.getName == experimentName)
    expOpt match {
      case Some(exp) => exp.getExperimentId
      case None => client.createExperiment(experimentName)
    }
  }
}

File: examples/mlflow_utils.py
'''MLFlow utilities'''

import mlflow

def get_or_create_experiment_id(experiment_name):
    tracking_uri = mlflow.tracking.get_tracking_uri()
    client = mlflow.tracking.MlflowClient(tracking_uri)

    exp = client.get_experiment_by_name(experiment_name)
    if exp is None:
        experiment_id = mlflow.create_experiment(experiment_name)
    else:
        experiment_id = exp.experiment_id
    print("experiment_name={} experiment_id={}".format(experiment_name,experiment_id))
    return experiment_id
